agent/src/softwarecontaineragent.cpp

```diff
@@ -192,17 +205,11 @@ ContainerID SoftwareContainerAgent::findSuitableId()
 
 SoftwareContainerAgent::SoftwareContainerPtr SoftwareContainerAgent::makeSoftwareContainer(const ContainerID containerID)
 {
-    std::string bridgeIp = m_config->getStringValue(ConfigDefinition::SC_GROUP,
-                                                    ConfigDefinition::SC_BRIDGE_IP_KEY);
-    int netmaskBits = m_config->getIntValue(ConfigDefinition::SC_GROUP,
-                                            ConfigDefinition::SC_BRIDGE_NETMASK_BITS_KEY);
-
     log_debug() << "Created container with ID :" << containerID;
 
     auto container = SoftwareContainerPtr(new SoftwareContainer(m_softwarecontainerWorkspace,
                                                                 containerID,
-                                                                bridgeIp,
-                                                                netmaskBits));
+                                                                m_containerConfig));
```

Comment:
Shouldn't this be a deep copy of the config, rather than a shared pointer at this point? If I'm reading this correctly, this will mean that the following would be true:

1) Call create(enableWriteBuffer = true) = SC-0
2) Call create(enableWriteBuffer = false) = SC-1

Wouldn't the second call change the enableWriteBuffer setting in SC-0 to false as well? That is not desired behaviour I think, even if we only use those values in the create step.